oecophylla/util/parse.py

```diff
@@ -0,0 +1,260 @@
+
+    Returns
+    -------
+    dict of list of str
+       Samples with a list of their forward and reverse files.
+
+    Notes
+    -----
+    This function is very illumina specific, strictly assuming that
+    the fwd reads are denoted by `R1` and the reverse reads are denoted
+    by `R2`.
+    """
+    sample_reads_dict = {}
+
+    samples = list(df[sample_col].unique())
+
+    for s in samples:
+        fwd = df.loc[(df[sample_col] == s) & (df[read_col] == 'R1'), file_col]
+        rev = df.loc[(df[sample_col] == s) & (df[read_col] == 'R2'), file_col]
+        f_fps = sorted(list(fwd.values))
+        r_fps = sorted(list(rev.values))
+
+        sample_reads_dict[s] = {'forward': [os.path.join(seq_dir, x) for x in f_fps],
+                                'reverse': [os.path.join(seq_dir, x) for x in r_fps]}
+
+    return sample_reads_dict
+
+
+def extract_sample_paths(seq_dir):
+    """ Obtain the sample paths.
+
+    Parameters
+    ----------
+    seq_dir : str
+       Input directory containing all of the sample files.
+
+    Returns
+    -------
+    dict of list of str
+       Samples with a list of their forward and reverse files.
+    """
+    fps = os.listdir(seq_dir)
+
+    files_df = illumina_filenames_to_df(fps)
+    sample_reads_dict = extract_sample_reads(files_df, seq_dir)
+
+    return sample_reads_dict
+
+
+def add_filter_db(sample_fp_dict, db_fp, samples = None,
+                  filter_col='filter_db'):
+    """ Add in a database for filtering out contaminant reads.
+
+    This is useful for filtering out reads from other sources
+    such as human DNA.
+
+    Parameters
+    ----------
+    sample_fp_dict : dict of list of str
+       Samples with a list of their forward and reverse files.
+    db_fp : str
+       Filepath of the database.
+    samples : list
+       List of sample names.
+    filter_col : str
+       Keyword name for the filtering database (default: 'filter_db')
+
+    Returns
+    -------
+    dict of list of str
+       Samples with a list of their forward and reverse files.
+    """
+
+    if samples is None:
+        samples = set(sample_fp_dict)
+
+    samples_dict = copy.deepcopy(sample_fp_dict)
+
+    for s in samples_dict:
+        if s in samples:
+            samples_dict[s][filter_col] = db_fp
+        elif filter_col in samples_dict[s]:
+            continue
+        else:
+            samples_dict[s][filter_col] = None
+
+    return(samples_dict)
+
+# Option 2: read samples from sample sheet
+def read_sample_sheet(f, sep='\t', comment='#'):
+    """ Outputs a dataframe from a sample sheet
+
+    Parameters
+    ----------
+    f : str
+       File name for reading in the sample sheet information.
+    sep : str
+       Delimiter for parsing the pandas input.
+
+    Returns
+    -------
+    data_df : pd.DataFrame
+       DataFrame containing the sample sheet information.
+
+    Note
+    ----
+    An example of a sample sheet would be as follows
```

Comment:
I'm left with a feeling of wanting more...

Reply:
That file can be found here
https://github.com/biocore/oecophylla/blob/master/test_data/example_sample_sheet.txt